scripts/hud/synchronizer.js

```diff
@@ -31,32 +31,58 @@ class Synchronizer {
 	static firstPanelWidth = this.maxSegmentWidth;
 	static syncGain = 10; // scale how fast the bars move
 	static altColor = 'rgba(0, 0, 0, 0.0)';
+	static lastVel = { x: 0, y: 0 };
+	static lastYaw = 0;
 
 	static onLoad() {
 		this.initializeSettings();
 
 		if (this.mom_hud_synchro_stat_mode) this.onJump(); // show stats if enabled
+
+		this.airSpeed = 30;
+		this.runSpeed = GameInterfaceAPI.GetSettingFloat('sv_maxspeed');
+		this.acceleration = GameInterfaceAPI.GetSettingFloat('sv_accelerate');
+		this.airAcceleration = GameInterfaceAPI.GetSettingFloat('sv_airaccelerate');
+		this.stopSpeed = GameInterfaceAPI.GetSettingFloat('sv_stopspeed');
+		this.friction = GameInterfaceAPI.GetSettingFloat('sv_friction');
+		this.tickInterval = MomentumMovementAPI.GetTickInterval();
+		this.maxGroundSpeed = this.getMaxGroundSpeed();
 	}
 
 	static onUpdate() {
 		const lastMoveData = MomentumMovementAPI.GetLastMoveData();
 		const lastTickStats = MomentumMovementAPI.GetLastTickStats();
 		const wishDir = lastMoveData.wishdir;
-		const velDir = this.getNormal(MomentumPlayerAPI.GetVelocity(), 0.001);
+
+		const vel = MomentumPlayerAPI.GetVelocity();
+		const velDir = this.getNormal(vel, 0.001);
+		const lastSpeed = this.getSize(this.lastVel);
+		this.lastVel = vel;
 
 		const bValidWishMove = this.getSize(wishDir) * this.getSize(velDir) > 0.001;
 		const strafeRight = (bValidWishMove ? 1 : 0) * lastTickStats.strafeRight;
 		const direction = this.mom_hud_synchro_dynamic_enable === 1 ? strafeRight : 1;
 		const flip = this.mom_hud_synchro_flip_enable === 1 ? -1 : 1;
 
+		const yaw = MomentumPlayerAPI.GetAngles().y;
+		const yawDelta = bValidWishMove ? this.wrapValueToRange(this.lastYaw - yaw, -180, 180) * this.deg2rad : 0;
+		this.lastYaw = yaw;
+
+		lastMoveData.isGrounded = lastMoveData.acceleration !== this.airAcceleration; // lastMoveData.moveStatus doesn't work
+		lastMoveData.maxspeed = lastMoveData.isGrounded ? this.runSpeed : this.airSpeed;
```

Comment:
Is `maxspeed` just some extra stuff you're adding to this object, or is it a value originally on `MomentumPlayerAPI`? If the latter, can just ignore this comment. But if it is a new value, I think it's clearer to keep this as a separate `const maxSpeed = lastMoveData.grounded ? ...` and add as an argument to `findOptimalDelta`. Since JS is untyped it's often difficult keeping track of what properties exist on what object, so best to avoid just chucking extra ones onto them.

Reply:
Yeah, `maxspeed` is a value originally on `MomentumMovementAPI.GetLastMoveData()` but it, along with `moveStatus` and `friction` apparently aren't hooked up and only work on Defrag.

Reply:
I decided to make `onGround` its own variable and then pass that into `findOptimalDelta` instead of `lastMoveData` where we then determine `maxSpeed`. Though this shouldn't even matter because `lastMoveData` should have these values but they only work on Defrag. Should I make a separate issue for that?

Reply:
Hey, I should have some time later today to review. The missing `lastMoveData` fields should be hooked up in engine. I can work on making those engine changes. Feel free to create an issue.